pages/glossary/semantic-role.md

```diff
@@ -9,4 +9,17 @@ A _semantic [role](https://www.w3.org/TR/wai-aria-1.1/#dfn-role)_ is a semantic
 
 Valid semantic roles are defined by standards. For web content and applications these are defined within the various [WAI-ARIA specifications](https://act-rules.github.io/glossary/#wai-aria-specifications).
 
-The semantic role of an element is its [explicit semantic role](#explicit-role) if it has any, otherwise, the [implicit semantic role](#implicit-role) is used.
+The _semantic role_ of an element which is [included in the accessibility tree][] is the role this element has in the accessibility tree. Elements which are not [included in the accessibility tree][] have no semantic role.
```

Comment:
So elements that aren't in the accessibility tree by definition have no role? That's a substantial change to one of our most used definitions. It means every time way say "has semantic role of X and is included in the accessibility tree" the second part becomes redundant. I'm hesitant about making that change. We'd have to update quite a few rules, make this "included in the accessibility tree" bit hidden away in the definition, which I don't like, and it feels a little unintuitive to me. For example, it would mean nothing can ever have a semantic role of "none". That's different from how semantic role is used by the ARIA working group.

I think we can add this in without such a substantial change to the definition.

```suggestion
The _semantic role_ of an element is its [explicit semantic role](#explicit-role) if it has any, otherwise, the [implicit semantic role](#implicit-role) is used. If the element has an [explicit semantic role[#explicit-role] of "presentation" or "none", but the element is, or would be, included in the accessibility tree when it is not [hidden](https://www.w3.org/TR/wai-aria-1.2/#dfn-hidden), the [implicit semantic role][#implicit-role] is used instead of the [explicit semantic role][#explicit-role].
```

Reply:
Very good point.
I wanted to try an avoid going through the full case study on our definition but this seems unavoidable…